### DISCUSSION TITLE
Currently, Does the Sheet component have a way to disable autofocus on open?

### DISCUSSION THREAD

**Discussion by oriolmontcreus** [huntabyte/shadcn-svelte#1342] [Help] [answered]
I've already tried:  ` openFocus={null}` . 
```svelte
  <Sheet
  openFocus={null}
  open={selectedProject !== null}
>
    <SheetContent>
``` 
but it isnt working. In shadcn from react it would look something like this: `onOpenAutoFocus={(e) => e.preventDefault()}`. 
The docs of the Sheet component already reproduces this autofocus behaviour.

**Comment by nikitaiavdeev** [comment]
You should be able to do this to disable open autofocus

```
<Sheet.Content
  onOpenAutoFocus={(e) => {
    e.preventDefault();
  }}
>
```

**Reply by oriolmontcreus** [reply]
'onOpenAutoFocus' doesnt exist for neither the Sheet.Content or Sheet.Root.
![image](https://github.com/user-attachments/assets/bb8e0b5d-0c9e-490e-b45c-87f05aecb137)
This is my current implementation (autofocus not disabled).
```svelte
<script lang="ts">
  import * as Sheet from "$lib/components/ui/sheet/index.js";
<Sheet.Root open={p !== null}>
</script>

  <Sheet.Content>
    <Sheet.Header>
      <Sheet.Title>{p.title}</Sheet.Title>
      <Sheet.Description>
        <div class="flex items-center">
          <UserAvatar
            src={p.creator.avatar}
            username={p.creator.username}
            className="mr-2"
          />
          <span>Created by {p.creator.username}</span>
        </div>
      </Sheet.Description>
    </Sheet.Header>
         <span>Data here</span>
  </Sheet.Content>
</Sheet.Root>

```

**Comment by nikitaiavdeev** [comment]
It does on my side. Are you using next for bits-ui and shandcs-svelte?

![image](https://github.com/user-attachments/assets/b71cd34a-44c6-4b1c-a048-5771731e59a6)

**Reply by oriolmontcreus** [reply]
Oh thats weird. Yes im using shadcn-svelte, a clean install with `npx shadcn-svelte@latest add sheet` and its still not working. For reference im using Astro and Svelte, and yes I have bits-ui installed. 
![image](https://github.com/user-attachments/assets/5c8cd0a4-086c-41bd-98e1-a36cc3391124)
![image](https://github.com/user-attachments/assets/414e2b5a-655b-463a-9056-a811b5308c81)

**Reply by nikitaiavdeev** [reply]
I'm sorry, I assumed you're using -next version. It looks like "onOpenAutoFocus" option was added for the -next version and isn't available in -latest.

https://next.bits-ui.com/docs/components/dialog#open-focus

**Reply by oriolmontcreus** [reply, marked as answer]
Yeah.. thats what I thought would happen... Well, I have come with a "solution".
```svelte
<script>
  function disableAutofocus(): FocusTarget {
    return document.querySelector(".disable-autofocus") as FocusTarget;
  }
</script>

<Sheet open={p !== null} openFocus={() => disableAutofocus()}>
  <SheetContent>
    <!-- Disable autofocus -->
    <span
      class="disable-autofocus"
      tabindex="-1"
    />
...
``` 

This is nasty but works, I guess I will use this.
I appreciate the help @nikitaiavdeev =)
